common/credentialFactory.go

```diff
@@ -166,6 +173,120 @@ func CreateBlobClientFromServiceClient(blobURLParts blob.URLParts, client *blobs
 	return blobClient, nil
 }
 
+func CreateContainerClient(blobURL string, credInfo *CredentialInfo, options *container.ClientOptions, credOpOptions *CredentialOpOptions) (*container.Client, error) {
```

Comment:
Ideally I don't want so much CreateXXXClient code. I will follow up this PR with some code to clean this up a bit and parse the URL, get service client, then get the appropriate container/blob client so we isolate this SAS/OAuth/SharedKey switch statement to one method